### PR TITLE
glimpse: Add version 0.2.0

### PR DESCRIPTION
**Pull request by bb010g**
I don't know how to handle the `0.2` instances better in autoupdates (matching on GitHub version results?), but here's a patch anyways. (It's still correctly automatic on point releases, like the upcoming 0.2.1.)

---
This change is [Reviewable](https://reviewable.io/reviews/lukesampson/scoop-extras/4842)

### CODE REVIEW

File: bucket/glimpse.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            "64bit": {
+                "extract_dir": "PFiles\\Glimpse Image Editor\\Glimpse $version (64-bit)",
+                "url": "https://github.com/glimpse-editor/Glimpse/releases/download/v$version/glimpse-$version-x64.msi"
+            },
+            "32bit": {
+                "extract_dir": "PFiles\\Glimpse Image Editor\\Glimpse $version",
+                "url": "https://github.com/glimpse-editor/Glimpse/releases/download/v$version/glimpse-$version-i686.msi"
+            }
+        }
+    }
```
Comment (Ash258) [inline bucket/glimpse.json:75]: Nonsense to run autoupdate in Excavator when there is hardcoded version in executables

```suggestion
    ]
```

File: bucket/glimpse.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
+{
+    "version": "0.2.0",
+    "description": "Glimpse is an open source image editor based on the GNU Image Manipulation Program. The goal is to experiment with new ideas and expand the use of free software.",
```
Comment (Ash258) [inline bucket/glimpse.json:3]:
```suggestion
    "description": "Image editor based on the GNU Image Manipulation Program",
```

File: bucket/glimpse.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    "homepage": "https://glimpse-editor.github.io/",
+    "license": "GPL-3.0-or-later",
+    "architecture": {
+        "64bit": {
+            "extract_dir": "PFiles\\Glimpse Image Editor\\Glimpse 0.2.0 (64-bit)",
```
Comment (Ash258) [inline bucket/glimpse.json:8]: Move this property after hash property

File: bucket/glimpse.json [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            "url": "https://github.com/glimpse-editor/Glimpse/releases/download/v0.2.0/glimpse-0.2.0-x64.msi",
+            "hash": "420dbbe11fef2a1d741083eb72f99732b9aec620d1d8274f9fb7a30e650030ff"
+        },
+        "32bit": {
+            "extract_dir": "PFiles\\Glimpse Image Editor\\Glimpse 0.2.0",
```
Comment (Ash258) [inline bucket/glimpse.json:13]: Move this property after hash property

File: bucket/glimpse.json [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            "url": "https://github.com/glimpse-editor/Glimpse/releases/download/v0.2.0/glimpse-0.2.0-i686.msi",
+            "hash": "2163403ca13d27be399b7fc1348719b2a9a4ddc4bc8c4b00aa8edd608bacfdbd"
+        }
+    },
+    "installer": {
+        "script": [
+            "Push-Location \"$dir\"",
```
Comment (Ash258) [inline bucket/glimpse.json:20]: Do not change directory

File: bucket/glimpse.json [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+            "if ($architecture -eq '64bit') {",
+            "   $defpath = \"`nPATH=`${gimp_installation_dir}\\bin;`${gimp_installation_dir}\\32\\bin`n\"",
+            "} else {",
+            "   $defpath = \"`nPATH=`${gimp_installation_dir}\\bin`n\"",
+            "}",
+            "$defenv = Get-Content -Raw 'lib\\glimpse\\2.0\\environ\\default.env'",
+            "$defenv += $defpath",
+            "$defenv | Set-Content 'lib\\glimpse\\2.0\\environ\\default.env'",
+            "Pop-Location"
```
Comment (Ash258) [inline bucket/glimpse.json:29]:
```powershell
$defpath = 'PATH=${gimp_installation_dir}\\bin'
if ($architecture -eq '64bit') { $defpath = "$defpath;`${gimp_installation_dir}\\32\\bin" }
($defpath, (Get-Content -Raw "$dir\lib\..\default.env)) -join "`n""
Set-Content "$dir\lib\...default.env" -Encoding Ascii
```

File: bucket/glimpse.json [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+            "hash": "420dbbe11fef2a1d741083eb72f99732b9aec620d1d8274f9fb7a30e650030ff"
+        },
+        "32bit": {
+            "extract_dir": "PFiles\\Glimpse Image Editor\\Glimpse 0.2.0",
+            "url": "https://github.com/glimpse-editor/Glimpse/releases/download/v0.2.0/glimpse-0.2.0-i686.msi",
+            "hash": "2163403ca13d27be399b7fc1348719b2a9a4ddc4bc8c4b00aa8edd608bacfdbd"
+        }
+    },
+    "installer": {
+        "script": [
```
Comment (Ash258) [inline bucket/glimpse.json:19]: Why is this needed in first place?

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/glimpse.json (hunk 7 above)
Comment (bb010g) [reply bucket/glimpse.json:19]: This is a pruned variant of the GIMP's installer script, no longer dealing with files that aren't present. I'm not totally sure why theirs does what it does, and I did not test a no installer script variant.

File: bucket/glimpse.json (hunk 3 above)
Comment (bb010g) [reply bucket/glimpse.json:8]: Is there a linter or fixer script that can be ran while writing or editing manifests before submission?

File: bucket/glimpse.json (hunk 5 above)
Comment (bb010g) [reply bucket/glimpse.json:20]: This should also be fixed in `bucket/gimp.json` then.

File: bucket/glimpse.json (hunk 6 above)
Comment (bb010g) [reply bucket/glimpse.json:29]: This cleanup should also be applied to `bucket/gimp.json`.

**Comment by TrechNex** [comment]
If it helps, I (creator of the project you're porting) had a crack at this here: https://github.com/TrechNex/scoop-extras/blob/master/bucket/glimpse-editor.json

It stalled because I wasn't able to test it. But I've linked here in case it's helpful, as the GNU I.M.P JSON file was my starting point too. Our installer differs though because it's built with WiX Toolset, not InnoSetup.

**Comment by bb010g** [comment]
@TrechNex WiX seems to be pretty fine with the default MSI extraction. This was made with some cross-referencing of the WiX XML files, and I think it's all covered.

**Comment by pratikpc** [comment]
It seems the project has now been archived

**Review by issaclin32 — APPROVED** [review]
